### PR TITLE
Standardise return type of VerifyChallenge and coalesce errors into Error

### PR DESCRIPTION
**Pull request by aultimus**
Prefer strongly typed return types to interface{} and map[string]interface{}. Don't use two separate Error like return values.

### CODE REVIEW

**Review by jonatascastro12 — APPROVED** [review]
lgtm

**Comment by aultimus** [comment]
Need to release this as a major version update in semver terms as this is a breaking change. I see a test Makefile target but no instructions on usage. @maxdeviant do i run `make release` within this pr or merge this then run `make release afterwards  on main?

**Comment by maxdeviant** [comment]
> Need to release this as a major version update in semver terms as this is a breaking change. I see a test Makefile target but no instructions on usage. @maxdeviant do i run `make release` within this pr or merge this then run `make release afterwards on main?

The version bump should be done as a separate PR.

I'll DM you the internal docs we have on the SDK release process.

File: pkg/mfa/client.go [hunk 1]
```diff
@@ -263,11 +263,20 @@ func (c *Client) VerifyFactor(
 	return VerifyChallenge(ctx, opts)
 }
 
+type VerificationResponseError struct {
+	Code    string
+	Message string
+}
+
+func (r *VerificationResponseError) Error() string {
+	return fmt.Sprintf("code: %s. message: %v", r.Code, r.Message)
```
Comment (drewfradette) [inline pkg/mfa/client.go:272]: Adding more context to the message would be nice probably?
```suggestion
	return fmt.Sprintf("mfa verification failed: %s (code: %s)", err.Message, err.Code)
```

**Comment by AshGodfrey** [comment]
I can go ahead and make these changes @drewfradette - I'm going to make the changes requested in #188 and merge into this, and then we can continue review + merge this PR to the 2.0 branch from there, I think?

File: pkg/mfa/client.go [hunk 2]
```diff
@@ -87,24 +97,34 @@ type EnrollResponse struct {
 	UpdatedAt string `json:"updated_at"`
 
 	// The type of request either 'sms' or 'totp'
-	Type string `json:"type"`
+	Type FactorType `json:"type"`
 
-	// Details of the totp response will be 'null' if using sms
-	Totp map[string]interface{} `json:"totp"`
+	// Details of the totp response will be 'null' if using sms\
```
Comment (drewfradette) [inline pkg/mfa/client.go:102]:
```suggestion
	// Details of the totp response will be 'null' if using sms
```

**Review by drewfradette — APPROVED** [review]
one stray `\` but 🚀

**Pull request merged**